src/components/ErrorBoundary/ErrorBoundary.tsx

```diff
@@ -0,0 +1,83 @@
+import { Box } from '@chakra-ui/react'
+import { TRPCClientError } from '@trpc/client'
+import { TRPC_ERROR_CODE_KEY } from '@trpc/server/rpc'
+import { Component } from 'react'
+import { ErrorBoundaryProps, ErrorBoundaryState } from './ErrorBoundary.types'
+import { TRPCWithErrorCodeSchema } from '../../utils/error'
+import { UnexpectedErrorCard } from './UnexpectedErrorCard'
+import { CALLBACK_URL_KEY } from '~/constants/params'
+
+class ErrorBoundary extends Component<ErrorBoundaryProps, ErrorBoundaryState> {
+  constructor(props: ErrorBoundaryProps) {
+    super(props)
+
+    // Define a state variable to track whether is an error or not
+    this.state = { hasError: false }
+  }
+  static getDerivedStateFromError(error: unknown) {
+    // Update state so the next render will show the fallback UI
+    return { hasError: true, error }
+  }
+  componentDidCatch(error: Error) {
+    // You can use your own error logging service here
+    console.log({ error })
```

Comment:
nit: prefer `console.error` if we are even outputting the error.  Might want to change the shape too and just pass in the error as-is:

```ts
console.error(error)
```